### PR TITLE
Structure and wording update to Readme

### PR DESCRIPTION
**Pull request by TKAB**
A restructuring of the Readme to make it more clear that there are two main cases with a slightly differing configuration and more explanation of the workflow.

### CODE REVIEW

**Review by harp79 — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -10,30 +10,46 @@ This project is a [_fastlane_](https://github.com/fastlane/fastlane) plugin. To
...
-| Supported Platforms | ios, android |
-|---------------------|--------------|
+For Bitcode-enabled iOS apps we recommend to let the plugin handle upload of the app to App Store Connect and download of the dSYM files.
 
-## Is your app Bitcode enabled?
-> Only applies for apps distributed via the AppStore or TestFlight.
 
-If your app is bitcode enabled, then the dSYMs that are generated during the Xcode build are **_not_** the dSYMs you want to upload to Dynatrace. This is because Apple recompiles the application on their servers, generating new dSYM files in the process. These newly generated dSYM files need to be downloaded from *App Store Connect*, then processed and uploaded to Dynatrace.
+## Usage
+To get started, first, ask your Dynatrace administrator for a an API token - it can be generated in the global settings in "Integration > Dynatrace API". The token needs the permission 'Mobile symbolication file management' and is used by the plugin to obtain permission to upload the symbol and mapping files to Dynatrace.
```
Comment (cynicer) [inline README.md:24]: This only applies to the v1 API, is there a way to do this through v2 already? Maybe this should be mentioned here.

**Review by cynicer — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (TKAB) [reply README.md:24]: No, v1 API only.

Comment (Landsi) [reply README.md:24]: "for (a) an API token"

File: README.md [hunk 2]
```diff
@@ -10,30 +10,46 @@ This project is a [_fastlane_](https://github.com/fastlane/fastlane) plugin. To
...
-This plugin allows you to decode and upload symbol files (iOS) or just upload obfuscation mapping files (Android) to Dynatrace. You can also use it to first download your latest dSYM files from App Store Connect if you use Bitcode.
+The Dynatrace fastlane plugin manages uploading symbol files (iOS) or obfuscation mapping files (Android) to the Dynatrace cluster. Symbol and mapping files are used to make reported stack traces human-readable. The plugin also allows to download the latest dSYM files from App Store Connect, which enables full automation of the mobile app deployment, and the pre-processing of dSYM files, a step that is necessary for the Dynatrace cluster to be able to symbolicate.
 
-## Action: `dynatrace_process_symbols`
+The plugin provides a single action `dynatrace_process_symbols`. The configuration depends on whether the app is (A) iOS and Bitcode-enabled or (B) iOS and not Bitcode-enabled or an Android app.
 
-| Supported Platforms | ios, android |
-|---------------------|--------------|
+For Bitcode-enabled iOS apps we recommend to let the plugin handle upload of the app to App Store Connect and download of the dSYM files.
```
Comment (Landsi) [inline README.md:20]: the plugin only handles dlsym download, processing & upload to Dynatrace, correct? The upload of the app is not part of this and usually done with normal fastlane actions?

File: README.md [hunk 3]
```diff
@@ -10,30 +10,46 @@ This project is a [_fastlane_](https://github.com/fastlane/fastlane) plugin. To
...
+## A) Bitcode-enabled iOS apps
+This is the right option if your app is distributed via Apple's App Store or TestFlight and Bitcode-enabled. For all other cases look at option B below.
+
+Background: If your app is Bitcode-enabled, then the dSYMs that are generated during the Xcode build are **not** the dSYMs that need to be uploaded to Dynatrace. Apple recompiles the application on their servers, generating new dSYM files in the process. These newly generated dSYM files need to be downloaded from *App Store Connect*, processed and uploaded to Dynatrace.
+
+### Automatically downloading dSYMs
+
+To fully automate the following five step workflow, add the snippets below to the respective files and fill in the placeholders:
+
+1. Upload build to App Store Connect
```
Comment (Landsi) [inline README.md:43]: again, the plugin doesn't upload the app - only waits for processing to finish

so the plugin should be called after uploading the app to apple (with normal fastlane action)

File: README.md [hunk 4]
```diff
@@ -89,21 +117,20 @@ dynatrace_process_symbols(
...
+When the plugin is used to download symbols from *App Store Connect* automatically (`downloadDsyms: true`) valid login App Store Connect credentials with access to the dSYM files are required. The preferred method of doing so is by setting the `FASTLANE_USER` and `FASTLANE_PASSWORD` environment variables to their respective values.
 
-[Apple announced](https://github.com/fastlane/fastlane/discussions/17655) that 2-Factor-Authentication for the *AppStore Connect* API will be enforced starting February 2021. This limits the ability to automate the symbol processing, because it will most likely involve manual interaction, which is not suitable for CI automation. The only workaround at this point in time is to pre-generate a session and cache it in CI.
+Apple announced that 2-Factor-Authentication for the *App Store Connect* API will be enforced starting February 2021. This [limits the ability to automate the symbol processing](https://github.com/fastlane/fastlane/discussions/17655), because it will most likely involve manual interaction, which is not suitable for CI automation. The only workaround at this point in time is to pre-generate a session and cache it in CI.
```
Comment (Landsi) [inline README.md:123]: This already happened. The fastlane session is currently the only way to (semi-)automate this, as described below. 

We need to watch out for when dsym APIs with an API Key are available from Apple & fastlane - which will hopefully happen some time soon..

**Review by Landsi — CHANGES_REQUESTED** [review]
The plugin doesn't upload the app. Other than that I do like the new structure!

File: README.md (hunk 2 above)
Comment (TKAB) [reply README.md]: Didn't realize that. It's fixed in the new commit.

**Comment by TKAB** [comment]
I commited a fix that addresses the issues that were pointed out. Could you please re-review?

**Review by Landsi — APPROVED** [review]

File: README.md [hunk 5]
```diff
@@ -45,18 +60,30 @@ dynatrace_process_symbols(
...
 ```
 
+#### Waiting time between app upload and dSYM file download
+There is a waiting time after the application has finished uploading to App Store Connect until the dSYM files are ready to be downloaded. The Dynatrace fastlane plugin waits and downloads the symbol files if setting the `waitForDsymProcessing` is true and a waiting period is provided via `waitForDsymProcessingTimeout`. We recommend 1800 seconds (30 mins) as the default waiting time. In our exprience this is sufficiently long for the processing to happen. If this duration is not long enough it needs to be increased. 
+
+> Note: this timeout is the **maximum** waiting time. If the symbol files are ready sooner, the plugin will continue to the download and will not wait for the whole duration of the timeout.
+
+
+## B) Not Bitcode-enabled iOS apps or Android apps
+If one of the following options is true, then this option is the right one for you:
```
Comment (jarekorlowski) [inline README.md:78]:
```suggestion
If at least one of the following conditions is true, you should follow this approach:
```

File: README.md [hunk 6]
```diff
@@ -89,21 +116,20 @@ dynatrace_process_symbols(
...
 | debugMode                    | Enable debug logging.                                                                                                                                                 | false          |
 
-## AppStore Connect Two-Factor-Authentication
-When the plugin is used to download symbols from *AppStore Connect* automatically (`downloadDsyms: true`)  valid login credentials to an account with permissions to access the dSYM files are required. The preferred method of doing so is by setting the `FASTLANE_USER` and `FASTLANE_PASSWORD` environment variables to their respective values.
+## App Store Connect Two-Factor-Authentication
+When the plugin is used to download symbols from *App Store Connect* automatically (`downloadDsyms: true`) valid login App Store Connect credentials with access to the dSYM files are required. The preferred method of doing so is by setting the `FASTLANE_USER` and `FASTLANE_PASSWORD` environment variables to their respective values.
```
Comment (jarekorlowski) [inline README.md:120]:
```suggestion
When the plugin is used to download symbols from *App Store Connect* automatically (`downloadDsyms: true`),  valid App Store Connect credentials with access to the dSYM files are required. The preferred method of doing so is by setting the `FASTLANE_USER` and `FASTLANE_PASSWORD` environment variables to their respective values.
```

File: README.md [hunk 7]
```diff
@@ -89,21 +116,20 @@ dynatrace_process_symbols(
...
 ) under **spaceauth**.
 
 You can generate a session by running `fastlane spaceauth -u user@email.com` on your machine and copy the output into an environment variable `FASTLANE_SESSION` on the target system (e.g. CI).
 
-### NOTE
-- Session is only valid in the "region" you create it. If you CI is in a different geographical location the authentication might fail.
-
-- Generated sessions are valid up to one month. Apple's API doesn't specify details about that, so it will only be visible by a failing build.
+> Note: 
+> - Session is only valid for the "region" you created it in. If you CI is in a different geographical location the authentication might fail
```
Comment (jarekorlowski) [inline README.md:131]:
```suggestion
> - Session is only valid for the "region" you created it in. If your CI is in a different geographical location, the authentication might fail
```

File: README.md [hunk 8]
```diff
@@ -10,30 +10,45 @@ This project is a [_fastlane_](https://github.com/fastlane/fastlane) plugin. To
 fastlane add_plugin dynatrace
 ```
 
-### Dynatrace Managed
-If the installation is on version 1.195 or earlier the Symbolication Client has to be manually download and specified (`dtxDssClientPath`), else it's fetched and updated automatically. A matching version can be downloaded manually with this link [https://api.mobileagent.downloads.dynatrace.com/sprint-latest-dss-client/xyz](https://api.mobileagent.downloads.dynatrace.com/sprint-latest-dss-client/xyz) by replacing `xyz` with the 3-digit sprint version of your Dynatrace Managed installation.
+⚠️  Apple enabled two-factor authentication in a way that interferes with a fully automated workflow. The workaround requires manual interaction see section 'App Store Connect Two-Factor-Authentication' below for details.
```
Comment (jarekorlowski) [inline README.md:13]:
```suggestion
⚠️  The way Apple introduced the two-factor authentication interferes with a fully automated workflow. The workaround requires manual interaction. For more information, see **App Store Connect Two-Factor-Authentication** section below for details.
```

File: README.md [hunk 9]
```diff
@@ -10,30 +10,45 @@ This project is a [_fastlane_](https://github.com/fastlane/fastlane) plugin. To
...
+## Usage
+To get started, first, ask your Dynatrace administrator for an API token - it can be generated in the global settings in "Integration > Dynatrace API". The token needs the permission 'Mobile symbolication file management' and is used by the plugin to obtain permission to upload the symbol and mapping files to Dynatrace.
 
-### Important
-There is a time gap between the application being uploaded to App Store Connect and the dSYM files to be ready. So **_we have to introduce some "wait" time in the CI to accomodate for this_**. You can do this by setting the `waitForDsymProcessing` to true. Unfortunately, Apple does not specify how long this time is. We recommend 1800 seconds (30 mins) as this is usually enough for the symbols are ready for download. You can increase this timeout if needed (`waitForDsymProcessingTimeout`). 
+Add the action `dynatrace_process_symbols` to your Fastfile, see further below for all configuration options and a default configuration.
```
Comment (jarekorlowski) [inline README.md:26]:
```suggestion
Add the action `dynatrace_process_symbols` to your Fastfile. You'll find all the configuration options and a default configuration later in the readme.
```

File: README.md [hunk 10]
```diff
@@ -10,30 +10,45 @@ This project is a [_fastlane_](https://github.com/fastlane/fastlane) plugin. To
...
 
-> Notice that this timeout is only the **maximum** waiting time. If the symbol files are ready sooner, it will continue processing and will not wait for the whole duration of the timeout.
+Now, when you run fastlane, the Dynatrace plugin will manage the symbol files of your app as configured.
+
+### Dynatrace Managed (1.195 and earlier)
+For cluster versions 1.195 and earlier the Dynatrace application 'Symbolication Client' has to be downloaded manually and explicitly specified (`dtxDssClientPath`). For all cluster versions above 1.195 it is fetched and updated automatically. A matching version can be downloaded manually with this link [https://api.mobileagent.downloads.dynatrace.com/sprint-latest-dss-client/xyz](https://api.mobileagent.downloads.dynatrace.com/sprint-latest-dss-client/xyz) by replacing `xyz` with the 3-digit sprint version of your Dynatrace Managed installation.
```
Comment (jarekorlowski) [inline README.md:31]:
```suggestion
For cluster versions 1.195 and earlier, the Dynatrace application 'Symbolication Client' has to be downloaded manually and explicitly specified (`dtxDssClientPath`). For all cluster versions above 1.195 it is fetched and updated automatically. A matching version can be downloaded manually with this link [https://api.mobileagent.downloads.dynatrace.com/sprint-latest-dss-client/xyz](https://api.mobileagent.downloads.dynatrace.com/sprint-latest-dss-client/xyz) by replacing `xyz` with the 3-digit sprint version of your Dynatrace Managed installation.
```

File: README.md [hunk 11]
```diff
@@ -10,30 +10,45 @@ This project is a [_fastlane_](https://github.com/fastlane/fastlane) plugin. To
...
+
+
+## A) Bitcode-enabled iOS apps
+This is the right option if your app is distributed via Apple's App Store or TestFlight and Bitcode-enabled. For all other cases look at option B below.
```
Comment (jarekorlowski) [inline README.md:35]:
```suggestion
This is the right approach if your app is distributed via Apple's App Store or TestFlight and Bitcode-enabled. For all other cases, follow the approach B below.
```

File: README.md [hunk 12]
```diff
@@ -10,30 +10,45 @@ This project is a [_fastlane_](https://github.com/fastlane/fastlane) plugin. To
...
+
+Background: If your app is Bitcode-enabled, then the dSYMs that are generated during the Xcode build are **not** the dSYMs that need to be uploaded to Dynatrace. Apple recompiles the application on their servers, generating new dSYM files in the process. These newly generated dSYM files need to be downloaded from *App Store Connect*, processed and uploaded to Dynatrace.
+
+### Automatically downloading dSYMs
+
+To fully automate the following five step workflow, add the snippets below to the respective files and fill in the placeholders. Uploading the app the App Store Connect is a necessary prerequisite and either handled manually or by fastlane directly:
```
Comment (jarekorlowski) [inline README.md:41]:
```suggestion
To fully automate the following five-step workflow, add the snippets below to the respective files and fill in the placeholders. Uploading the app the App Store Connect is a necessary prerequisite and either handled manually or by fastlane directly:
```

**Comment by TKAB** [comment]
@Landsi @cynicer changes are approved. You're good to merge!

**Pull request merged**